distsql: NPE when asked to log plan diagram for stats query

I'm running with `logPlanDiagram` hacked to true, and I got (rarely) the following crash. My line `distsql_running.go:332` corresponds to the following. Notice that the query being planned is a special one, about stats computation.

https://github.com/cockroachdb/cockroach/blob/38d069a84f83bec461c451cdcde8e073756f5e7a/pkg/sql/distsql_running.go#L328

@RaduBerinde , mind passing along?

```
E200115 22:14:19.355248 8935 util/log/crash_reporting.go:207  [n2] a panic has occurred!
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2895053]

goroutine 8935 [running]:
panic(0x3c2b000, 0x6af2cc0)
        /usr/local/go/src/runtime/panic.go:722 +0x2c2 fp=0xc005b24608 sp=0xc005b24578 pc=0x791eb2
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc00027b9e0, 0x494ea40, 0xc0050e6b40)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:182 +0x11f fp=0xc005b24668 sp=0xc005b24608 pc=0x1251bff
runtime.call32(0x0, 0x42aceb0, 0xc005b25f78, 0x1800000018)
        /usr/local/go/src/runtime/asm_amd64.s:539 +0x3b fp=0xc005b24698 sp=0xc005b24668 pc=0x7c080b
panic(0x3c2b000, 0x6af2cc0)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2 fp=0xc005b24728 sp=0xc005b24698 pc=0x791da2
runtime.panicmem(...)
        /usr/local/go/src/runtime/panic.go:199
runtime.sigpanic()
        /usr/local/go/src/runtime/signal_unix.go:394 +0x3ec fp=0xc005b24758 sp=0xc005b24728 pc=0x7a7acc
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc000b71600, 0xc005603320, 0xc005356ab0, 0xc005b24d60, 0xc0005f3600, 0xc000124de0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:332 +0xa33 fp=0xc005b24c28 sp=0xc005b24758 pc=0x2895053
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).planAndRunCreateStats(0xc000b71600, 0x494eb00, 0xc0057417a0, 0xc000124de0, 0xc005603320, 0xc005356ab0, 0xc0002b7ea0, 0xc0056698e0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_plan_stats.go:256 +0x276 fp=0xc005b24e40 sp=0xc005b24c28 pc=0x2891d26
github.com/cockroachdb/cockroach/pkg/sql.(*createStatsResumer).Resume.func2(0x494eb00, 0xc00534ee10, 0xc005356ab0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/create_stats.go:380 +0x21b fp=0xc005b24ef0 sp=0xc005b24e40 pc=0x29791eb
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1(0x494eb00, 0xc00534ee10, 0xc005356ab0, 0x49d6b00, 0xc005af2900)
        /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:717 +0x43 fp=0xc005b24f28 sp=0xc005b24ef0 pc=0x1340623
github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).exec(0xc005356ab0, 0x494eb00, 0xc00534ee10, 0xc005b25278, 0xc005356ab0, 0xc0056692e0)
        /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:751 +0xd9 fp=0xc005b25218 sp=0xc005b24f28 pc=0x133c4c9
github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn(0xc0008dc580, 0x494eb00, 0xc00534ee10, 0xc0059373a8, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:716 +0xd1 fp=0xc005b25298 sp=0xc005b25218 pc=0x1332041
github.com/cockroachdb/cockroach/pkg/sql.(*createStatsResumer).Resume(0xc005199f20, 0x494eb00, 0xc00534ee10, 0x40fbbe0, 0xc000124d00, 0xc004bc37a0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/create_stats.go:371 +0x34f fp=0xc005b25bb8 sp=0xc005b25298 pc=0x285c78f
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resume.func1(0x494eb00, 0xc00534ee10)
        /go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:662 +0x36d fp=0xc005b25ea8 sp=0xc005b25bb8 pc=0x16ccd3d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc00027b9e0, 0x494ea40, 0xc0050e6b40, 0xc00519a7e0, 0x1e, 0x0, 0x0, 0xc0051a9540)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:322 +0x152 fp=0xc005b25fa0 sp=0xc005b25ea8 pc=0x1254522
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc005b25fa8 sp=0xc005b25fa0 pc=0x7c2521
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:317 +0x131
```

--- a/pkg/sql/distsql_running.go
+++ b/pkg/sql/distsql_running.go
@@ -324,7 +324,7 @@ func (dsp *DistSQLPlanner) Run(
 	if logPlanDiagram {
 		log.VEvent(ctx, 1, "creating plan diagram")
 		var stmtStr string
-		if planCtx.planner != nil {
+		if planCtx.planner != nil && planCtx.planner.stmt != nil {
 			stmtStr = planCtx.planner.stmt.String()
 		}
 		_, url, err := execinfrapb.GeneratePlanDiagramURL(stmtStr, flows, false /* showInputTypes */)

